File: genefunk/src/main/java/de/hip/pnp/genefunk/GeneFunkGenomeType.java
package de.hip.pnp.genefunk;

import javax.persistence.Id;

enum GeneFunkGenomeType {
    UNKNOWN(-1),
    Engineered(0),
    MUTT(1),
    OPTIMIZED(2),
    TRANSHUMAN(3);
    @Id
    int value;

    GeneFunkGenomeType(int value) {
        this.value = value;
    }

    public static GeneFunkGenomeType valueOf(int value) {
        GeneFunkGenomeType[] values = GeneFunkGenomeType.values();
        for (GeneFunkGenomeType type : values) {
            if (type.getValue() == value) return type;
        }
        return GeneFunkGenomeType.UNKNOWN;
    }

    public int getValue() {
        return this.value;
    }
}

File: genefunk/src/main/kotlin/de/hipp/pnp/genefunk/GeneFunkGenomeType.kt
package de.hipp.pnp.genefunk

import jakarta.persistence.Id

enum class GeneFunkGenomeType(
    @Id var value: Int,
) {
    UNKNOWN(-1),
    ENGINEERED(0),
    MUTTS(1),
    OPTIMIZED(2),
    TRANSHUMAN(3),
    ;

    companion object {
        fun valueOf(value: Int): GeneFunkGenomeType {
            val values = entries.toTypedArray()
            for (type in values) {
                if (type.value == value) return type
            }
            return UNKNOWN
        }
    }
}
